### PR TITLE
feat add invitation banner for landing page

### PR DESCRIPTION
**Pull request by adam-strzelec**
## Description

Add colony invitation banner for landing page

## Testing

- Run storybook
- Open `Colony Invitation Banner`

## Diffs

**New stuff** ✨

Screenshot 2024-10-22 at 18 43 47

Screenshot 2024-10-22 at 19 03 50

Main issue - https://github.com/JoinColony/colonyCDapp/issues/3392

### CODE REVIEW

**Review by mmioana — CHANGES_REQUESTED** [review]
What a nice component @adam-strzelec 🙌 

Design wise seem to be looking as expected
Screenshot 2024-10-24 at 10 59 48
Screenshot 2024-10-24 at 10 59 50
Screenshot 2024-10-24 at 10 59 53

Left a veeeery tiny refactoring comment 😁 
But nice job 🎉

File: src/components/frame/LandingPage/ColonyInvitationBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {formatText({ id: 'landingPage.invitationBanner.title' })}
+      </h1>
+      <span className="rounded-xl bg-blue-100 px-3 py-1 text-sm font-medium text-blue-400">
+        {formatText(
+          { id: 'landingPage.invitationBanner.remaining' },
+          { remaining: coloniesRemaining },
+        )}
+      </span>
+
+      <p className="mt-2 text-sm font-normal">
```
Comment (mmioana) [inline src/components/frame/LandingPage/ColonyInvitationBanner.tsx:40]: And I think we should also apply `text-gray-700`

**Review by mmioana — CHANGES_REQUESTED** [review]
Sorry @adam-strzelec on a second look I found some small mismatches with Figma

File: src/components/frame/LandingPage/ColonyInvitationBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  return (
+    <div className="w-full max-w-[31.25rem] rounded-2xl border px-6 py-6">
+      <div className="flex h-10 w-10 items-center justify-center rounded-md border">
+        <SmileySticker size={24} />
+      </div>
+      <h1 className="my-2.5 heading-4">
+        {formatText({ id: 'landingPage.invitationBanner.title' })}
+      </h1>
+      <span className="rounded-xl bg-blue-100 px-3 py-1 text-sm font-medium text-blue-400">
+        {formatText(
```
Comment (bassgeta) [inline src/components/frame/LandingPage/ColonyInvitationBanner.tsx:36]: Could we move all i18n strings into this component?

**Review by bassgeta — CHANGES_REQUESTED** [review]
Another one smashed out of the park! Really nice, and a really simple component to look at.
I've just left a comment if we could move the i18n messages into this component and then we're good to go!
image

**Review by arrenv — CHANGES_REQUESTED** [review]
@adam-strzelec Thank you for the PR, and your work on the new landing page. Mostly looks good.

What I noticed is that the ticket icon is responsive, however, should be fixed. When using the full expected invite link, is aligns to the top and get squished. I have also exaggerated this with a really long invite link as well. 

### With the provided string
image

### With the expected string
image

### With an exaggerated string
image

**Review by arrenv — APPROVED** [review]
@adam-strzelec Thank you, looks good now!

File: src/components/frame/LandingPage/ColonyInvitationBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+      <p className="mt-2 text-sm font-normal text-gray-700">
+        <FormattedMessage
+          {...(hasColoniesRemaining
+            ? MSG.descriptionRemaining
+            : MSG.descriptionNoRemaining)}
+        />
+      </p>
+      {hasColoniesRemaining && (
+        <div className="mt-2 flex flex-col gap-2 rounded-md bg-gray-50 px-3 py-3">
```
Comment (rumzledz) [inline src/components/frame/LandingPage/ColonyInvitationBanner.tsx:74]: **[Nitpick alert]:** Instead of `px-3 py-3`, use `p-3`

Comment (rumzledz) [reply src/components/frame/LandingPage/ColonyInvitationBanner.tsx:74]: This should be `gap-4` (16px) according to the Figma designs:

gap-4

File: src/components/frame/LandingPage/ColonyInvitationBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  inviteLink,
+  coloniesRemaining,
+}: ColonyInvitationBannerProps) => {
+  const { handleClipboardCopy, isCopied } = useCopyToClipboard();
+
+  const hasColoniesRemaining = coloniesRemaining > 0;
+
+  return (
+    <div className="w-full max-w-[31.25rem] rounded-2xl border px-6 py-6">
+      <div className="flex h-10 w-10 items-center justify-center rounded-md border">
```
Comment (rumzledz) [inline src/components/frame/LandingPage/ColonyInvitationBanner.tsx:53]: The border radius should be `4px`, at the moment it's `6px`

**Review by rumzledz — CHANGES_REQUESTED** [review]
Hey @adam-strzelec I've just realised that this is the first time I'm speaking to you so hello! 👋 😂 

Great job on this PR so far! I left comments in areas that need to be tweaked, but I wanted to highlight this one:

The Copy link button should follow the Figma design system. Currently, when you click it, the only thing that changes is the SVG:

copylink

It should actually turn to green and the copy should say "Link copied". You'll find the design system here:

https://www.figma.com/design/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?node-id=1801-464&t=jofkV0FxoYS1PGK6-4

We have a `completed` value for the `mode` prop if you wanna experiment with that:

```js
<Button mode="completed" />
```

**Comment by rumzledz** [comment]
Sorry one more thing, you might want to handle the pluralisation of the word "colonies" as well. Should say "1 colony remaining"

image

**Comment by rumzledz** [comment]
Sorry one more thing, you might want to handle the pluralisation of the word "colonies" as well. Should say "1 colony remaining"

image

This should do the trick! 😄 

```js
defaultMessage: `{remaining} {remaining, plural, one {colony} other {colonies}} remaining`,
```

**Comment by adam-strzelec** [comment]
Hello @rumzledz Nice to meet you 🙂
I already added the corrections.

**Review by iamsamgibbs — APPROVED** [review]
Amazing job, this all looks great. Code is nicely structured and matches figma perfectly:

Screenshot 2024-10-28 at 11 45 08
Screenshot 2024-10-28 at 11 45 12
Screenshot 2024-10-28 at 11 45 42

**Pull request merged**